srcpkgs/godot/template

```diff
@@ -10,7 +10,7 @@ build_style=scons
 # Use builtin bullet for now as it's too old in repos (needs 2.89)
 # Toggle to not use builtin once bullet has been updated
 make_build_args="platform=x11 tools=yes target=release_debug dev=no progress=no
- pulseaudio=no builtin_bullet=false builtin_libpng=false builtin_libvpx=false
```

Comment:
I wouldn't delete this and just change it to yes. This PR doesn't make godot link libpulse, so I think there is a mistake that will become a failure if you change this to yes.

Reply:
I can try setting `pulseaudio=yes` and rebuilding, when I have a chance. Though I don't think Godot with Pulse has a hard dependency on having libpulseaudio installed, since if I `sudo mv /usr/lib/libpulse.so.0.24.0{,~}` then try running `godot` in a terminal, I see output:

```
libpulse.so.0: cannot open shared object file: No such file or directory
libpulse.so.0: cannot open shared object file: No such file or directory
```

followed by opening an ALSA audio stream instead (alsa-pipewire which constantly `ALSA lib pcm.c:8568:(snd_pcm_recover) underrun occurred`, presumably a bug but I didn't investigate what's wrong yet).

What's the best approach for declaring in a package that it needs either libpulse or libalsa, or it can't output audio (but still runs)?

Reply:
Setting pulseaudio=yes builds it with pulseaudio unconditionally enabled which is what we want. It's possible it dlopens libpulse in which case the dependency wouldn't be detected automatically.

Reply:
I checked the source code and yes, they are dlopening it, that means it will use libpulse if you have it on your system, use alsa if you don't and the missing dependency is fine.

Reply:
Godot has multiple audio drivers selected at runtime, and `pulseaudio=yes` *enables* the Pulse backend as a runtime *option* without turning off ALSA as an option if Pulse is absent:

```
godot --audio-driver test
Unknown audio driver 'test', aborting.
Valid options are 'PulseAudio', 'ALSA' and 'Dummy'.
```

And I think `pulseaudio=yes` is set by default: https://github.com/godotengine/godot/blob/3.5.1-stable/platform/x11/detect.py#L77

Should I add `pulseaudio=yes` or omit it altogether (which the flatpak build does, and enables PulseAudio)?

Reply:
I don't really have a preference